source/ncengine/graphics2/diligent/DiligentEngineWin32.cpp

```diff
@@ -76,6 +85,7 @@ DiligentEngine::DiligentEngine(const config::GraphicsSettings& graphicsSettings,
                     throw nc::NcError("Failed to create the D3D12 swapchain.");
                 }
 
+                m_shaderFactory = MakeShaderFactory(*pFactoryD3D12, *m_pDevice);
```

Comment:
I think we want to check for null and throw here for `m_shaderFactory` as well, in each of the API/Platform combos

Reply:
That's our own type/function. It won't return null, and it throws if it fails to create a `IShaderSourceInputStreamFactory`.